src/Microsoft.TemplateEngine.Abstractions/Installer/IInstaller.cs

```diff
@@ -38,5 +39,13 @@ public interface IInstaller
         Task<IReadOnlyList<IManagedTemplatesSourceUpdate>> GetLatestVersions(IEnumerable<IManagedTemplatesSource> sources);
 
         Task<IReadOnlyList<InstallResult>> UpdateAsync(IEnumerable<IManagedTemplatesSourceUpdate> sources);
+
+        /// <summary>
+        /// Creates list of autocomplete candidates for <see cref="InstallRequest.Identifier"/>.
+        /// </summary>
+        /// <param name="textSoFar">Text that was typed by user so far.</param>
+        /// <param name="token">Calling code might call this on every key press, hence it is good idea to respect <see cref="CancellationToken"/></param>
+        /// <returns>List of valid <see cref="InstallRequest.Identifier"/>s.</returns>
+        Task<IReadOnlyList<string>> GetAutocompleteAsync(string textSoFar, CancellationToken token);
```

Comment:
Is `List` absolutely needed here? Can we do lighter collection?

Reply:
We probably need to considering results limit - we don't want to get all possible options - should we read it from GlobalSettings?